playbooks/openstack/inventory.py

```diff
@@ -124,11 +124,10 @@ def build_inventory():
     '''Build the dynamic inventory.'''
     cloud = shade.openstack_cloud()
 
-    # TODO(shadower): filter the servers based on the `OPENSHIFT_CLUSTER`
-    # environment variable.
+    cluster_name = os.getenv('OPENSHIFT_CLUSTER', 'openshift-cluster')
     cluster_hosts = [
         server for server in cloud.list_servers()
-        if 'metadata' in server and 'clusterid' in server.metadata]
+        if 'metadata' in server and 'clusterid' in server.metadata and server.metadata['clusterid'] == cluster_name ]
```

Comment:
Actually, upon closer inspection this won't work; the openshift_openstack_clusterid is distinct from the cluster name. We use an environment variable for the cluster name precisely because it's not possible to coordinate parameters between the playbooks and inventory.py.

@tomassedovic Any suggestions here?

Reply:
I tried to take the same approach as documented when using a different cluster ID.

https://github.com/openshift/openshift-ansible/blob/master/playbooks/openstack/configuration.md#stack-name-configuration

Say we set OPENSHIFT_CLUSTER as 'devops-1' for instance. Exporting it and running against the inventory will capture 'devops-1' as the name. Previously, the heat template would set it as the full DNS domain name, which I reckoned was an oversight. I'd want my cluster ID to be 'devops-1', not 'devops-1.my.domain.name.'

In our case, we are not making a subdomain for each cluster. All entries are in the same zone so we override the openshift_openstack_full_dns_domain param. With the old heat template, all cluster ID metadata would be the same since they're all under the same full DNS domain.